types/styled-components/index.d.ts

```diff
@@ -74,13 +74,10 @@ export type StyledComponentProps<
     A extends keyof any
 > = WithOptionalTheme<
     Omit<
-        OmitOverlapping<
-            ReactDefaultizedProps<
-                C,
-                React.ComponentPropsWithRef<C>
-            >,
-            O
-        > & O,
+        ReactDefaultizedProps<
```

Comment:
I realized looking back here that I likely initially mis-ordered the application here, we'd still want `defaultProps` to apply `Partial` after re-picking the correct types.